download_node_csv.py

```diff
@@ -30,78 +34,126 @@ class City(Enum):
+    parser.add_argument('cookies_path', type=Path, help='path to cookie')
+    return parser.parse_args()
+
+def parse_nodes(r: str):
+    nodes = []
+    for node in literal_eval(r):
+        nodes.append(Node(lat=node[0], lon=node[1], names=f'{node[3]} ({str(node[2])[-3:]})'))
+    return nodes
+
+def enum_to_cityname(city: City):
+    return city.name.title().replace('_', ' ')
+
+def cache_filename(city: City):
+    return city.name.lower()
 
-def file_name(city: City):
-    return str(city)[5:].lower()
+@dataclass
+class CityGrid:
+    nelng: float
+    nelat: float
+    swlng: float
+    swlat: float
 
-print("Choose a city:")
-for i, c in enumerate(City):
-    print(f"{i + 1}. {name(c)}")
+    def __post_init__(self):
+        for k in fields(self):
+            value = getattr(self, k.name)
+            if not isinstance(value, k.type):
+                setattr(self, k.name, k.type(value))
 
-try:
-    city = list(City)[int(input("\nChoice: ")) - 1]
-except:
-    print(f"Invalid input: Number must be <= {len(City)}")
-    sys.exit()
+    def __hash__(self):
+        return hash((self.nelng, self.nelat, self.swlng, self.swlat))
 
-if   city == City.YORK:        coordinates = [-79.3829, 43.7206, -79.5560, 43.6424]
-elif city == City.WROCLAW:     coordinates = [17.078224311098552, 51.13988879756559,  17.00226573206399,  51.09263199227115 ]
-elif city == City.KRAKOW:      coordinates = [19.979436306324942, 50.08859858611942,  19.90053987336441,  50.034804024531525]
-elif city == City.ROME:        coordinates = [12.519622013860925, 41.91439535724936,  12.443743029831694, 41.85439499689079 ]
-elif city == City.VENICE:      coordinates = [12.357666134722393, 45.451386491714715, 12.316599314442499, 45.42077976598716 ]
-elif city == City.FOLKESTONE:  coordinates = [1.2028963028344322, 51.11176465501126,  1.1199095563368644, 51.05639602006997 ]
-elif city == City.MEAFORD:     coordinates = [-80.49028489574928, 44.754231277837675, -80.94388807383417, 44.44134084860639 ]
-else:                          coordinates = [-79.20, 43.8, -79.556, 43.61]
 
-[start_nelng, start_nelat, start_swlng, start_swlat] = coordinates
+CityGrids = {
+    City.YORK:        CityGrid(-79.3829, 43.7206, -79.5560, 43.6424),
+    City.WROCLAW:     CityGrid(17.078224311098552, 51.13988879756559,  17.00226573206399,  51.09263199227115 ),
+    City.KRAKOW:      CityGrid(19.979436306324942, 50.08859858611942,  19.90053987336441,  50.034804024531525),
+    City.ROME:        CityGrid(12.519622013860925, 41.91439535724936,  12.443743029831694, 41.85439499689079 ),
+    City.VENICE:      CityGrid(12.357666134722393, 45.451386491714715, 12.316599314442499, 45.42077976598716 ),
+    City.FOLKESTONE:  CityGrid(1.2028963028344322, 51.11176465501126,  1.1199095563368644, 51.05639602006997 ),
+    City.MEAFORD:     CityGrid(-80.49028489574928, 44.754231277837675, -80.94388807383417, 44.44134084860639 ),
+    City.OLD_TORONTO: CityGrid(-79.20, 43.8, -79.556, 43.61)
```

Comment:
This doesn't duplicate the `else` case on the left. Would need to add:
```py
    City.NORTH_YORK:  CityGrid(-79.20, 43.8, -79.556, 43.61),
    City.EAST_YORK:   CityGrid(-79.20, 43.8, -79.556, 43.61),
    City.ALL_TORONTO: CityGrid(-79.20, 43.8, -79.556, 43.61)
```

Reply:
good catch, fixed!